### ISSUE TITLE
Determine immutability for const type parameters

### ISSUE THREAD

**Issue by bjb568** [eslint-functional/eslint-plugin-functional#827]
## Suggestion

Example:

```ts
(<const T>(f: () => T) => f)(() => ({ foo: 1 }));
```

Currently this results in: Return type should have an immutability of at least "Immutable" (actual: "Mutable"). However, the type parameter uses the const keyword so the code is equivalent to the following, which is correctly treated as immutable.

```ts
(<T>(f: () => T) => f)(() => ({ foo: 1 } as const));
```

(Not sure if this is a suggestion for this repo or for [is-immutable-type](https://github.com/RebeccaStevens/is-immutable-type) so I made it here.)

**Comment by RebeccaStevens** [comment]
It seems to be an upstream issue in `ts-api-utils`. I've created an issue over there, I'll close this one.
